### PR TITLE
QPID-8500:[Broker-J]Introduce flag to disable coalescing commiter thread

### PR DESCRIPTION
**Pull request by Dedeepya-T**
A context variable is added into BDB HA to give an option to disable coalescing committer thread. For backward compatibility, by default it will be set to false.

### CODE REVIEW

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 1]
```diff
@@ -1111,7 +1129,7 @@ public void onNoMajority()
             }
         };
 
-        ReplicatedEnvironmentConfiguration masterConfig = createReplicatedEnvironmentConfiguration(TEST_NODE_NAME, TEST_NODE_HOST_PORT, false);
+        ReplicatedEnvironmentConfiguration masterConfig = createReplicatedEnvironmentConfiguration(TEST_NODE_NAME, TEST_NODE_HOST_PORT, false,false);
```
Comment (alex-rufous) [inline bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:1132]: please revert

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 2]
```diff
@@ -1160,7 +1178,7 @@ public void onNoMajority()
...
```
Comment (alex-rufous) [inline bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:1181]: please revert

**Review by alex-rufous — CHANGES_REQUESTED** [review]

File: bdbstore/src/main/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacade.java [hunk 3]
```diff
@@ -135,6 +137,11 @@
      */
     private final int _executorShutdownTimeout;
 
+    /**
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/main/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacade.java:140]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 4]
```diff
@@ -186,7 +187,7 @@ public void tearDown() throws Exception
     @Test
     public void testClose() throws Exception
     {
-        ReplicatedEnvironmentFacade ef = createMaster();
+        ReplicatedEnvironmentFacade ef = createMaster(false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:190]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 5]
```diff
@@ -199,7 +200,7 @@ public void testOpenDatabaseReusesCachedHandle() throws Exception
     {
         DatabaseConfig createIfAbsentDbConfig = DatabaseConfig.DEFAULT.setAllowCreate(true);
 
-        EnvironmentFacade ef = createMaster();
+        EnvironmentFacade ef = createMaster(false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:203]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 6]
```diff
@@ -217,7 +218,7 @@ public void testOpenDatabaseWhenFacadeIsNotOpened() throws Exception
...
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:221]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 7]
```diff
@@ -234,19 +235,19 @@ public void testOpenDatabaseWhenFacadeIsNotOpened() throws Exception
     @Test
     public void testGetGroupName() throws Exception
     {
-        assertEquals("Unexpected group name", TEST_GROUP_NAME, createMaster().getGroupName());
+        assertEquals("Unexpected group name", TEST_GROUP_NAME, createMaster(false).getGroupName());
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:238]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 8]
```diff
@@ -234,19 +235,19 @@ public void testOpenDatabaseWhenFacadeIsNotOpened() throws Exception
...
     }
 
     @Test
     public void testGetNodeName() throws Exception
     {
-        assertEquals("Unexpected group name", TEST_NODE_NAME, createMaster().getNodeName());
+        assertEquals("Unexpected group name", TEST_NODE_NAME, createMaster(false).getNodeName());
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:244]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 9]
```diff
@@ -234,19 +235,19 @@ public void testOpenDatabaseWhenFacadeIsNotOpened() throws Exception
...
     }
 
     @Test
     public void testLastKnownReplicationTransactionId() throws Exception
     {
-        ReplicatedEnvironmentFacade master = createMaster();
+        ReplicatedEnvironmentFacade master = createMaster(false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:250]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 10]
```diff
@@ -256,21 +257,21 @@ public void testLastKnownReplicationTransactionId() throws Exception
     @Test
     public void testGetNodeHostPort() throws Exception
     {
-        assertEquals("Unexpected node host port", TEST_NODE_HOST_PORT, createMaster().getHostPort());
+        assertEquals("Unexpected node host port", TEST_NODE_HOST_PORT, createMaster(false).getHostPort());
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:260]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 11]
```diff
@@ -256,21 +257,21 @@ public void testLastKnownReplicationTransactionId() throws Exception
...
                             TEST_NODE_HELPER_HOST_PORT,
-                            createMaster().getHelperHostPort());
+                            createMaster(false).getHelperHostPort());
     }
 
     @Test
     public void testSetMessageStoreDurability() throws Exception
     {
-        ReplicatedEnvironmentFacade master = createMaster();
+        ReplicatedEnvironmentFacade master = createMaster(false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:274]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 12]
```diff
@@ -1016,7 +1034,7 @@ public void onNodeRolledback()
                 LOGGER.debug("onNodeRolledback in " + TEST_NODE_NAME);
                 _replicaRolledback.countDown();
             }
-        });
+        },false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:1037]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 13]
```diff
@@ -284,10 +285,27 @@ public void testSetMessageStoreDurability() throws Exception
...
     }
 
+    @Test
+    public void testSetMessageStoreDurabilityWithDisabledCoalescingSync() throws Exception
+    {
+        ReplicatedEnvironmentFacade master = createMaster(true);
+        assertEquals("Unexpected message store durability",
+                     new Durability(Durability.SyncPolicy.SYNC, Durability.SyncPolicy.NO_SYNC, Durability.ReplicaAckPolicy.SIMPLE_MAJORITY),
+                     master.getRealMessageStoreDurability());
+        assertEquals("Unexpected durability", TEST_DURABILITY, master.getMessageStoreDurability());
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:295]: Removed the duplicate!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 14]
```diff
@@ -284,10 +285,27 @@ public void testSetMessageStoreDurability() throws Exception
...
+                     new Durability(Durability.SyncPolicy.SYNC, Durability.SyncPolicy.WRITE_NO_SYNC, Durability.ReplicaAckPolicy.ALL),
+                     master.getRealMessageStoreDurability());
+        assertFalse("Coalescing sync committer is still running", master.isCoalescingSync());
+    }
+
     @Test
     public void testGetNodeState() throws Exception
     {
-        assertEquals("Unexpected state", State.MASTER.name(), createMaster().getNodeState());
+        assertEquals("Unexpected state", State.MASTER.name(), createMaster(false).getNodeState());
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:308]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 15]
```diff
@@ -284,10 +285,27 @@ public void testSetMessageStoreDurability() throws Exception
...
+    public void testSetMessageStoreDurabilityWithDisabledCoalescingSync() throws Exception
+    {
+        ReplicatedEnvironmentFacade master = createMaster(true);
+        assertEquals("Unexpected message store durability",
+                     new Durability(Durability.SyncPolicy.SYNC, Durability.SyncPolicy.NO_SYNC, Durability.ReplicaAckPolicy.SIMPLE_MAJORITY),
+                     master.getRealMessageStoreDurability());
+        assertEquals("Unexpected durability", TEST_DURABILITY, master.getMessageStoreDurability());
+        assertFalse("Unexpected coalescing sync", master.isCoalescingSync());
+
+        master.setMessageStoreDurability(Durability.SyncPolicy.SYNC, Durability.SyncPolicy.WRITE_NO_SYNC, Durability.ReplicaAckPolicy.ALL);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:298]: Changed!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 16]
```diff
@@ -296,7 +314,7 @@ public void testPriority() throws Exception
         final TestStateChangeListener masterListener = new TestStateChangeListener();
         final ReplicationGroupListener masterGroupListener = new NoopReplicationGroupListener();
 
-        ReplicatedEnvironmentConfiguration masterConfig = createReplicatedEnvironmentConfiguration(TEST_NODE_NAME, TEST_NODE_HOST_PORT, TEST_DESIGNATED_PRIMARY);
+        ReplicatedEnvironmentConfiguration masterConfig = createReplicatedEnvironmentConfiguration(TEST_NODE_NAME, TEST_NODE_HOST_PORT, TEST_DESIGNATED_PRIMARY, false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:317]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 17]
```diff
@@ -317,7 +335,7 @@ public void testDesignatedPrimary()  throws Exception
         final TestStateChangeListener masterListener = new TestStateChangeListener();
         final ReplicationGroupListener masterGroupListener = new NoopReplicationGroupListener();
 
-        ReplicatedEnvironmentConfiguration masterConfig = createReplicatedEnvironmentConfiguration(TEST_NODE_NAME, TEST_NODE_HOST_PORT, TEST_DESIGNATED_PRIMARY);
+        ReplicatedEnvironmentConfiguration masterConfig = createReplicatedEnvironmentConfiguration(TEST_NODE_NAME, TEST_NODE_HOST_PORT, TEST_DESIGNATED_PRIMARY,false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:338]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 18]
```diff
@@ -338,7 +356,7 @@ public void testElectableGroupSizeOverride() throws Exception
         final TestStateChangeListener masterListener = new TestStateChangeListener();
         final ReplicationGroupListener masterGroupListener = new NoopReplicationGroupListener();
 
-        ReplicatedEnvironmentConfiguration masterConfig = createReplicatedEnvironmentConfiguration(TEST_NODE_NAME, TEST_NODE_HOST_PORT, false);
+        ReplicatedEnvironmentConfiguration masterConfig = createReplicatedEnvironmentConfiguration(TEST_NODE_NAME, TEST_NODE_HOST_PORT, false,false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:359]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 19]
```diff
@@ -361,7 +379,7 @@ public void testElectableGroupSizeOverride() throws Exception
     @Test
     public void testReplicationGroupListenerHearsAboutExistingRemoteReplicationNodes() throws Exception
     {
-        ReplicatedEnvironmentFacade master = createMaster();
+        ReplicatedEnvironmentFacade master = createMaster(false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:382]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 20]
```diff
@@ -403,7 +421,7 @@ public void onReplicationNodeAddedToGroup(ReplicationNode node)
         };
 
         TestStateChangeListener stateChangeListener = new TestStateChangeListener();
-        ReplicatedEnvironmentFacade replicatedEnvironmentFacade = addNode(stateChangeListener, listener);
+        ReplicatedEnvironmentFacade replicatedEnvironmentFacade = addNode(stateChangeListener, listener,false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:424]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 21]
```diff
@@ -455,7 +473,7 @@ public void onReplicationNodeRemovedFromGroup(ReplicationNode node)
...
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:476]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 22]
```diff
@@ -514,7 +532,7 @@ public void onNodeState(ReplicationNode node, NodeState nodeState)
...
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:535]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 23]
```diff
@@ -540,7 +558,7 @@ public void onNodeState(ReplicationNode node, NodeState nodeState)
     public void testRemoveNodeFromGroup() throws Exception
     {
         TestStateChangeListener stateChangeListener = new TestStateChangeListener();
-        ReplicatedEnvironmentFacade environmentFacade = addNode(TEST_NODE_NAME, TEST_NODE_HOST_PORT, true, stateChangeListener, new NoopReplicationGroupListener());
+        ReplicatedEnvironmentFacade environmentFacade = addNode(TEST_NODE_NAME, TEST_NODE_HOST_PORT, true, stateChangeListener, new NoopReplicationGroupListener(),false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:561]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 24]
```diff
@@ -567,7 +585,7 @@ public void testRemoveNodeFromGroupTwice() throws Exception
                                                                 TEST_NODE_HOST_PORT,
                                                                 true,
                                                                 stateChangeListener,
-                                                                new NoopReplicationGroupListener());
+                                                                new NoopReplicationGroupListener(),false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:588]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 25]
```diff
@@ -630,7 +648,7 @@ public void onNodeState(ReplicationNode node, NodeState nodeState)
         };
 
         TestStateChangeListener stateChangeListener = new TestStateChangeListener();
-        final ReplicatedEnvironmentFacade masterEnvironment = addNode(stateChangeListener, listener);
+        final ReplicatedEnvironmentFacade masterEnvironment = addNode(stateChangeListener, listener,false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:651]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 26]
```diff
@@ -661,7 +679,7 @@ public void onNodeState(ReplicationNode node, NodeState nodeState)
     @Test
     public void testCloseStateTransitions() throws Exception
     {
-        ReplicatedEnvironmentFacade replicatedEnvironmentFacade = createMaster();
+        ReplicatedEnvironmentFacade replicatedEnvironmentFacade = createMaster(false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:682]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 27]
```diff
@@ -706,7 +724,7 @@ else if (stateChangeEvent.getState() == State.UNKNOWN)
         };
 
         // make sure that node is re-elected as MASTER on second start-up
-        ReplicatedEnvironmentConfiguration config = createReplicatedEnvironmentConfiguration(TEST_NODE_NAME, TEST_NODE_HOST_PORT, TEST_DESIGNATED_PRIMARY);
+        ReplicatedEnvironmentConfiguration config = createReplicatedEnvironmentConfiguration(TEST_NODE_NAME, TEST_NODE_HOST_PORT, TEST_DESIGNATED_PRIMARY,false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:727]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 28]
```diff
@@ -761,7 +779,7 @@ public void stateChange(StateChangeEvent event) throws RuntimeException
                 }
             }
         };
-        ReplicatedEnvironmentFacade firstNode = addNode(stateChangeListener, new NoopReplicationGroupListener());
+        ReplicatedEnvironmentFacade firstNode = addNode(stateChangeListener, new NoopReplicationGroupListener(),false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:782]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 29]
```diff
@@ -791,7 +809,7 @@ public void stateChange(StateChangeEvent event) throws RuntimeException
             }
         };
         ReplicatedEnvironmentFacade thirdNode = addNode(TEST_NODE_NAME + "_2", node2NodeHostPort, TEST_DESIGNATED_PRIMARY,
-                                                        testStateChangeListener, new NoopReplicationGroupListener());
+                                                        testStateChangeListener, new NoopReplicationGroupListener(),false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:812]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 30]
```diff
@@ -823,7 +841,7 @@ public void stateChange(StateChangeEvent event) throws RuntimeException
                 }
             }
         };
-        ReplicatedEnvironmentFacade firstNode = addNode(stateChangeListener, new NoopReplicationGroupListener());
+        ReplicatedEnvironmentFacade firstNode = addNode(stateChangeListener, new NoopReplicationGroupListener(),false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:844]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java (hunk 2 above)
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:1181]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java (hunk 1 above)
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:1132]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 31]
```diff
@@ -1072,7 +1090,7 @@ public Transaction call() throws Exception
     @Test
     public void testReplicaWriteExceptionIsConvertedIntoConnectionScopedRuntimeException()  throws Exception
     {
-        ReplicatedEnvironmentFacade master = createMaster();
+        ReplicatedEnvironmentFacade master = createMaster(false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:1093]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 32]
```diff
@@ -1034,7 +1052,7 @@ public void onNodeRolledback()
     @Test
     public void testReplicaTransactionBeginsImmediately()  throws Exception
     {
-        ReplicatedEnvironmentFacade master = createMaster();
+        ReplicatedEnvironmentFacade master = createMaster(false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:1055]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 33]
```diff
@@ -993,7 +1011,7 @@ public void testNodeRolledback()  throws Exception
         LOGGER.debug("RESTARTING " + replicaName);
         // Restart the node2, making it primary so it becomes master
         TestStateChangeListener node2StateChangeListener = new TestStateChangeListener();
-        node2 = addNode(replicaName, replicaNodeHostPort, true, node2StateChangeListener, new NoopReplicationGroupListener());
+        node2 = addNode(replicaName, replicaNodeHostPort, true, node2StateChangeListener, new NoopReplicationGroupListener(),false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:1014]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 34]
```diff
@@ -967,7 +985,7 @@ public void testNodeRolledback()  throws Exception
         createConfig.setTransactional(true);
 
         TestStateChangeListener masterListener = new TestStateChangeListener();
-        ReplicatedEnvironmentFacade node1 = addNode(TEST_NODE_NAME, TEST_NODE_HOST_PORT, true, masterListener, new NoopReplicationGroupListener());
+        ReplicatedEnvironmentFacade node1 = addNode(TEST_NODE_NAME, TEST_NODE_HOST_PORT, true, masterListener, new NoopReplicationGroupListener(),false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:988]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 35]
```diff
@@ -945,7 +963,7 @@ public boolean onIntruderNode(ReplicationNode node)
                 return true;
             }
         };
-        ReplicatedEnvironmentFacade firstNode = createMaster(listener);
+        ReplicatedEnvironmentFacade firstNode = createMaster(listener,false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:966]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 36]
```diff
@@ -921,7 +939,7 @@ public void testPermittedNodeIsAllowedToConnect() throws Exception
         permittedNodes.add(node1NodeHostPort);
         firstNode.setPermittedNodes(permittedNodes);
 
-        ReplicatedEnvironmentConfiguration configuration =  createReplicatedEnvironmentConfiguration(TEST_NODE_NAME + "_1", node1NodeHostPort, false);
+        ReplicatedEnvironmentConfiguration configuration =  createReplicatedEnvironmentConfiguration(TEST_NODE_NAME + "_1", node1NodeHostPort, false,false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:942]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 37]
```diff
@@ -911,7 +929,7 @@ public void testSetPermittedNodes() throws Exception
     @Test
     public void testPermittedNodeIsAllowedToConnect() throws Exception
     {
-        ReplicatedEnvironmentFacade firstNode = createMaster();
+        ReplicatedEnvironmentFacade firstNode = createMaster(false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:932]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 38]
```diff
@@ -891,7 +909,7 @@ public void testBeginTransaction() throws Exception
     @Test
     public void testSetPermittedNodes() throws Exception
     {
-        ReplicatedEnvironmentFacade firstNode = createMaster();
+        ReplicatedEnvironmentFacade firstNode = createMaster(false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:912]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 39]
```diff
@@ -868,7 +886,7 @@ public void stateChange(StateChangeEvent event) throws RuntimeException
     @Test
     public void testBeginTransaction() throws Exception
     {
-        ReplicatedEnvironmentFacade facade = createMaster();
+        ReplicatedEnvironmentFacade facade = createMaster(false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:889]: Done!

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java [hunk 40]
```diff
@@ -854,7 +872,7 @@ public void stateChange(StateChangeEvent event) throws RuntimeException
         };
         String thirdNodeName = TEST_NODE_NAME + "_2";
         ReplicatedEnvironmentFacade thirdNode = addNode(thirdNodeName, node2NodeHostPort, TEST_DESIGNATED_PRIMARY,
-                                                        testStateChangeListener, new NoopReplicationGroupListener());
+                                                        testStateChangeListener, new NoopReplicationGroupListener(),false);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/replication/ReplicatedEnvironmentFacadeTest.java:875]: Done!

**Review by alex-rufous — CHANGES_REQUESTED** [review]
Please merge into a single commit. I will apply it to master

**Review by alex-rufous — APPROVED** [review]

**Pull request closed without merging**